checker/src/org/checkerframework/checker/lock/qual/PolyGuardedBy.java

```diff
@@ -0,0 +1,36 @@
+/*
+TODO: Implement the functionality for @PolyGuardedBy and uncomment this.
```

Comment:
I have not yet seen a user-written method that needs it, but `@PolyGuardedBy` covers scenarios that `@GuardedByUnknown` and `@GuardSatisfied` don't. For example:

    @PolyGuardedBy Object checkNotNull(@PolyGuardedBy Object o) {
        if (o == null) {
            saveChangesToDiskBeforePossibleNullPointerException();
        }
        return o;
    }

`@GuardedByUnknown` could be used if the method signature could be changed to `<T extends @GuardedByUnknown Object> T checkNotNull(T ref)`, but perhaps the user can't change the signature for some reason.
`@GuardSatisfied` cannot be used since the guard does not need to be satisfied.